tests/v2/validation/fleet/snapshot/snapshot_restore_existing_gitrepo_test.go

```diff
@@ -0,0 +1,263 @@
+		},
+	}
+
+	f.client, err = f.client.ReLogin()
+	require.NoError(f.T(), err)
+
+	userConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, userConfig)
+
+	clusterObject, _, _ := extensionscluster.GetProvisioningClusterByName(f.client, f.client.RancherConfig.ClusterName, fleet.Namespace)
+	if clusterObject != nil {
+		status := &provv1.ClusterStatus{}
+		err := steveV1.ConvertToK8sType(clusterObject.Status, status)
+		require.NoError(f.T(), err)
+
+		f.clusterID = status.ClusterName
+	} else {
+		f.clusterID, err = extensionscluster.GetClusterIDByName(f.client, f.client.RancherConfig.ClusterName)
+		require.NoError(f.T(), err)
+	}
+
+	podErrors := pods.StatusPods(f.client, f.clusterID)
+	require.Empty(f.T(), podErrors)
+
+	//may not be necessary since this isn't dynamic
+	logrus.Info("loading snapshot config")
+	f.clustersConfig = new(etcdsnapshot.Config)
+	config.LoadConfig(etcdsnapshot.ConfigurationFileKey, f.clustersConfig)
+	logrus.Info("Done")
+}
+
+func (f *FleetWithSnapshotTestSuite) TestSnapshotThenFleetRestore() {
+	snapshotRestoreAll := &etcdsnapshot.Config{
+		UpgradeKubernetesVersion:     "",
+		SnapshotRestore:              "all",
+		ControlPlaneConcurrencyValue: "15%",
+		ControlPlaneUnavailableValue: "3",
+		WorkerConcurrencyValue:       "20%",
+		WorkerUnavailableValue:       "15%",
+		RecurringRestores:            1,
+	}
+
+	tests := []struct {
+		name         string
+		etcdSnapshot *etcdsnapshot.Config
+	}{
+		{" Restore cluster config Kubernetes version and etcd", snapshotRestoreAll},
+	}
+
+	for _, tt := range tests {
+		testSession := session.NewSession()
+		defer testSession.Cleanup()
+		client, err := f.client.WithSession(testSession)
+		require.NoError(f.T(), err)
+
+		fleetVersion, err := fleet.GetDeploymentVersion(client, fleet.FleetControllerName, fleet.LocalName)
+		require.NoError(f.T(), err)
+
+		urlQuery, err := url.ParseQuery(fmt.Sprintf("labelSelector=%s=%s", "cattle.io/os", "windows"))
+		require.NoError(f.T(), err)
+
+		steveClient, err := client.Steve.ProxyDownstream(f.clusterID)
+		require.NoError(f.T(), err)
+
+		winsNodeList, err := steveClient.SteveType("node").List(urlQuery)
+		require.NoError(f.T(), err)
+
+		if len(winsNodeList.Data) > 0 {
+			f.fleetGitRepo.Spec.Paths = []string{fleet.GitRepoPathWindows}
+			f.fleetGitRepo.Name += "windows"
+			tt.name += "windows"
+		}
+
+		client, err = client.ReLogin()
+		require.NoError(f.T(), err)
+
+		f.Run(fleet.FleetName+" "+fleetVersion+tt.name, func() {
+			var isRKE1 = false
+
+			clusterObject, _, _ := extensionscluster.GetProvisioningClusterByName(client, client.RancherConfig.ClusterName, namespace)
+			if clusterObject == nil {
+				_, err := client.Management.Cluster.ByID(f.clusterID)
+				require.NoError(f.T(), err)
+
+				isRKE1 = true
+			}
+
+			containerTemplate := workloads.NewContainer(containerImage, containerImage, corev1.PullAlways, []corev1.VolumeMount{}, []corev1.EnvFromSource{}, nil, nil, nil)
```

Comment:
Deployments are part of the tests for snapshots. I didn't think it would make sense to not validate everything we normally do for snapshots. Rather I wanted to validate everything we do for snapshots and in addition, add fleet into the mix.

Reply:
I’m a bit confused on why deployments are a part of the snapshot tests. The primary goal of snapshot is to make sure that the resources persist before and after the snapshot, for which we're specifically validating the Fleet resources. In my opinion, including deployments isn't necessary since the tests are mainly to verify fleet resources are intact before and after a snapshot.

Reply:
we want to append to the existing snapshot test. The test is snapshot + fleet, so why would I change what is tested as part of snapshots outside of _adding_ tests for fleet?